src/containers/FilmFrontpage/MovieGrid.tsx

```diff
@@ -6,51 +6,129 @@
 import { GQLResourceTypeMoviesQuery, GQLResourceTypeMoviesQueryVariables } from "../../graphqlTypes";
 import { useGraphQuery } from "../../util/runQueries";
 
-const StyledSection = styled.section`
-  margin-left: ${spacing.normal};
-  margin-right: ${spacing.normal};
-  ${mq.range({ from: breakpoints.desktop })} {
-    margin-left: ${spacing.xlarge};
-    margin-right: ${spacing.xlarge};
-  }
-`;
+const StyledSection = styled("section", {
+  base: {
+    display: "flex",
+    justifyContent: "center",
+  },
+});
 
-const MovieListing = styled.div`
-  display: flex;
-  flex-wrap: wrap;
-  gap: ${spacing.normal};
-  margin: ${spacing.small} 0;
-`;
+const ColumnWrapper = styled("div", {
+  base: {
+    display: "flex",
+    flexDirection: "column",
+  },
+});
 
-const StyledFilmContentCard = styled(FilmContentCard)`
-  opacity: 0;
-  animation-fill-mode: forwards;
-  animation-name: fadeIn;
-  animation-duration: 300ms;
-  animation-delay: calc(var(--index) * 50ms);
-`;
+const MovieListing = styled("div", {
+  base: {
+    display: "grid",
+    gap: "medium",
+    gridTemplateColumns: "repeat(4, 1fr)",
+    desktopDown: { gridTemplateColumns: "repeat(3, 1fr)" },
+    tabletDown: { gridTemplateColumns: "repeat(2, 1fr)" },
+  },
+});
 
-const StyledHeading = styled(Heading)`
-  display: flex;
-  gap: ${spacing.small};
-  margin: ${spacing.xsmall} 0;
-`;
+const StyledFilmContentCard = styled(FilmContentCard, {
+  base: {
+    animationFillMode: "backwards",
+    animationName: "fade-in",
+    animationDuration: "300ms",
+    animationDelay: "calc(var(--index) * 50ms)",
+  },
+});
+
+const StyledHeading = styled(Heading, {
+  base: {
+    display: "flex",
+    gap: "small",
+    paddingBlockEnd: "xsmall",
+  },
+});
+
+const LoadingShimmer = () => {
+  return new Array(24).fill(0).map((_, index) => (
+    <Skeleton key={index}>
+      <StyledFilmContentCard
+        movie={{
+          id: `dummy-${index}`,
+          resourceTypes: [],
+          path: "",
+          title: "",
+        }}
+      />
+    </Skeleton>
+  ));
+};
 
 interface Props {
   resourceType: { id: string; name: string };
-  loadingPlaceholderHeight?: string;
 }
 
+const MovieGrid = ({ resourceType }: Props) => {
+  const { t, i18n } = useTranslation();
+  const resourceTypeMovies = useGraphQuery<GQLResourceTypeMoviesQuery, GQLResourceTypeMoviesQueryVariables>(
+    resourceTypeMoviesQuery,
+    {
+      variables: {
+        resourceType: resourceType.id,
+        language: i18n.language,
+      },
+    },
+  );
+  const [films, setFilms] = useState<GQLResourceTypeMoviesQuery | undefined>(undefined);
+  const [prevResourceType, setPrevResourceType] = useState(resourceType);
+  useEffect(() => {
+    if (prevResourceType !== resourceType) {
+      setFilms(undefined);
+      setPrevResourceType(resourceType);
+      return;
+    }
+    setFilms(resourceTypeMovies.data);
+  }, [prevResourceType, resourceType, resourceTypeMovies.data]);
```

Comment:
Gjorde dette for å triggre animasjonen hver gang man bytter (også om dataen er i cache).
Kanskje litt hacky, men den forrige viste animasjonen fra hvor mange kort som fantes i den forrige kategorien, og det føltes litt teitorama :thinking:

Reply:
Kom gjerne med forslag på hvordan gjøre dette bedre / penere om det finnes noen triks :woman_juggling:

Reply:
Jonas kom med triks irl :heart:

Reply:
Katrine så på